ui/app/adapters/config-ui/message.js

```diff
@@ -4,5 +4,26 @@
  */
 
 import ApplicationAdapter from '../application';
+import { encodePath } from 'vault/utils/path-encoding-helpers';
 
-export default class MessageAdapter extends ApplicationAdapter {}
+export default class MessageAdapter extends ApplicationAdapter {
+  getCustomMessagesUrl(id) {
+    let url = `${this.buildURL()}/config/ui/custom-messages`;
+
+    if (id) {
+      url = url + '/' + encodePath(id);
+    }
+    return url;
+  }
+
+  async query(store, type, query) {
+    const { authenticated } = query;
```

Comment:
What if query is false? Does the API error? You might want to consider a catch statement here and doing something with the error if it's a 404 (no response, e.g. no items in the list) vs anything else like a 403. There is a good example of this on the ldap library adapter.

Reply:
I originally had a try/catch here but the 404/403 error didn't bubble to the query. Since I'm using lazyPaginatedQuery, I did the try/catch in the route instead and the errors get surfaced in there. https://github.com/hashicorp/vault/pull/24133/files#diff-5a46be6bfd29de6b487be8b6242b62bdd861524c47cc1fb10eb4a876e00c129f